### DISCUSSION TITLE
Swap - sempre utile?

### DISCUSSION THREAD

**Discussion by Andrea-12345** [indomus/forum#132] [GENERALE - Discussioni generiche]
Salve.

Ho acquistato un Raspberry Pi4 con 8gb di RAM e sto seguendo le Vostre guide per "imparare" qualcosa e sperimentare.
Mi ritrovo però già da subito a farVi una domanda.

Da quanto ho capito, Swap è una partizione che viene utilizzata se saturiamo lo spazio della RAM.
Ha senso aumentare lo swap se abbiamo già 8 gb di ram?
È consigliabile aumentarla a priori?
Nel normale utilizzo, per Vostra esperienza, quanta RAM viene utilizzata?
L'ultima capisco sia una domanda un po' aleatoria: se uno fa grafica col pc avrà necessità diverse rispetto a chi scrive con word... ma almeno da avere un'idea di quanta ram utilizza un sistema di domotica "tipo", basato su Raspbian + Docker + Home Assistant Core.
Grazie.

**Comment by snakuzzo** [comment]
Ciao...ti dico...
Io fino a poco fa avevo un RPI3B (quindi 1GB di RAM) con un SSD "scrauso" da 32GB ed usavo questa roba con Docker...

HA Core
Mosquitto
Emby Server
JDownloader
Portainer
Openvpn server
Caddy
PiHole
Zigbee2mqtt

...mai avuto problemi.
La swap memory sul RPI l'ho adeguata in base a quanto puoi leggere online su decine di articoli, anche se di fatto nel mio caso non era realmente necessario

![20210411_121836.jpg](https://user-images.githubusercontent.com/7205371/114300425-8fafe600-9ac0-11eb-9e34-24a76c7de086.jpg)

L'utilizzo della RAM ovviamente dipende anche da come è scritta l'applicazione che stai usando, indipendentemente dallo scopo d'uso.

Nel caso specifico di HA non puoi sapere a priori la RAM che occupa. Dipende anche dalle integrazioni e custom component che hai

**Comment by Andrea-12345** [comment]
Ok. Chiaro, grazie.
Diciamo che nel mio caso non sarebbe necessario, ma comunque consigliato.
